subiquitycore/ui/form.py

```diff
@@ -504,13 +523,12 @@ def as_screen(self, focus_buttons=True, excerpt=None, narrow_rows=False):
             focus_buttons=focus_buttons, excerpt=excerpt,
             narrow_rows=narrow_rows)
 
+    def has_validation_error(self) -> bool:
+        """ Tells if any field is in error. """
+        return any(map(lambda f: f.is_in_error(), self._fields))
+
     def validated(self):
```

Comment:
I'm not sure I have any better ideas but I'm not sure that "validated" is a very useful name for this method. Can you think of a better one? :)

Reply:
Agreed. How do you feel about the following?

```python
def on_field_validated(self) -> None:
    """ Method to call when a field is validated. """
    self.refresh_done_button()
    
def refresh_done_button(self) -> None:
    """ Look for any error in the fields and enable/disable the Done button accordingly. """
    # here goes the body of `validated`
```

fields would call `on_field_validated` after undertaking validation.

Reply:
I did implement this change but I'm not too happy with it at the moment ; so I'm not pushing it yet.
The change in subiquity/ui/views/filesystem/guided.py proves that there are other scenarios where we'd want to refresh the done button (e.g., toggling a radio button); not just when a field undertakes validation.
I want to give it more thoughts and I will go with a follow up PR.